core/src/main/java/io/grpc/internal/ApplicationThreadDeframer.java

```diff
@@ -78,31 +79,43 @@ public void run() {
                 if (deframer.isClosed()) {
                   return;
                 }
+                Status s;
                 try {
-                  deframer.request(numMessages);
+                  s = deframer.request(numMessages);
                 } catch (Throwable t) {
-                  storedListener.deframeFailed(t);
+                  s = Status.UNKNOWN.withCause(t).withDescription(
+                      "unexpected error requesting messages");
+                }
+                if (!s.isOk()) {
+                  storedListener.deframeFailed(s);
                   deframer.close(); // unrecoverable state
                 }
               }
             }));
+    return Status.OK;
```

Comment:
One alternative (as adopted by AtomicStampedReference) is to pass in an array to propagate a second return value.   The returns could go back the way they were.  

Also, I could wrap status here in another type.

Reply:
Yeah, having a holder for an error would fix this issue. I'd suggest we'd just make a StatusHolder for the task. It would look like C error handling then.

Reply:
Done, PTAL.